src/core/frameworks/gql/generatePossibleTypes.js

```diff
@@ -0,0 +1,55 @@
+// Purpose: Generate possibleTypes.json to pass as parameter when initializing
+// InMemoryChache in frameworks/gql/index. Enables usage of fragments with
+// unions and interfaces in GraphQL queries.
+
+// Source Documentation: https://www.apollographql.com/docs/react/data/fragments/#using-fragments-with-unions-and-interfaces
+
+const fs = require('fs');
+const path = require('path');
+
+const fetch = require('cross-fetch');
+
+const GRAPH_API_URL = process.env.GRAPH_API_URL;
+
+const homeDirectory = path.resolve(process.cwd());
+const destinationDirectory = '/src/core/frameworks/gql/possibleTypes.json';
+const destinationPath = path.join(homeDirectory, destinationDirectory);
+
+fetch(`${GRAPH_API_URL}`, {
```

Comment:
can u add this to the `index.ts` file. doesnt really need its own file. Also can u call this when we init a new client in `getClient` instead of instantly.

Reply:
added. 

one caveat - I attempted to also saved the `possibleTypes` object to `frameworks/gql/queries.ts`, but that causes the `generateSubgraphTypes` function to throw the following error: 
`TypeError: fs__WEBPACK_IMPORTED_MODULE_0__.writeFile is not a function`

I've added TODO comments for the `generateSubgraphTypes` function.

Reply:
this code is in this file + index.js now. Where are you going to keep it?